syscall_intercept does not build with clang 5.0.0

GCC 7.2.0 works fine though

```
cmake .. -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++
make -j8
```

```
Scanning dependencies of target check_whitespace
Scanning dependencies of target syscall_intercept_base_clf
Scanning dependencies of target cpp_test
Scanning dependencies of target check_license_executable
Scanning dependencies of target test_clone_thread
Scanning dependencies of target syscall_intercept_base_asm
Scanning dependencies of target executable_with_syscall_pie
Scanning dependencies of target cstyle
[  0%] Building C object CMakeFiles/syscall_intercept_base_clf.dir/src/cmdline_filter.c.o
[  1%] Building CXX object CMakeFiles/cpp_test.dir/src/cpp_compile_test.cc.o
[  1%] Building C object CMakeFiles/check_license_executable.dir/utils/check_license/check-license.c.o
[  2%] Building C object test/CMakeFiles/test_clone_thread.dir/test_clone_thread.c.o
[  2%] Building ASM object CMakeFiles/syscall_intercept_base_asm.dir/src/intercept_template.s.o
[  3%] Building ASM object test/CMakeFiles/executable_with_syscall_pie.dir/executable_with_syscall.s.o
[  3%] Built target check_whitespace
Scanning dependencies of target hook_test_preload_o
[  4%] Building C object test/CMakeFiles/hook_test_preload_o.dir/hook_test_preload.c.o
[  5%] Building ASM object CMakeFiles/syscall_intercept_base_asm.dir/src/util.s.o
[  5%] Building C object CMakeFiles/cpp_test.dir/src/cpp_compile_mock.c.o
[  5%] Built target syscall_intercept_base_clf
[  6%] Linking ASM executable executable_with_syscall_pie
[  6%] Linking C executable test_clone_thread
Scanning dependencies of target hook_test
[  6%] Building C object test/CMakeFiles/hook_test.dir/hook_test.c.o
[  6%] Built target hook_test_preload_o
[  7%] Linking CXX executable cpp_test
[  8%] Building ASM object CMakeFiles/syscall_intercept_base_asm.dir/src/intercept_wrapper.s.o
Scanning dependencies of target pattern_jmps.out
[  9%] Linking C executable check_license_executable
[  9%] Built target executable_with_syscall_pie
[  9%] Built target cstyle
[ 10%] Building ASM object test/CMakeFiles/pattern_jmps.out.dir/pattern_jmps.out.s.o
Scanning dependencies of target pattern_jmps.in
Scanning dependencies of target pattern_rets.in
[ 10%] Built target test_clone_thread
[ 11%] Building ASM object test/CMakeFiles/pattern_rets.in.dir/pattern_rets.in.s.o
[ 12%] Building ASM object test/CMakeFiles/pattern_jmps.in.dir/pattern_jmps.in.s.o
Scanning dependencies of target pattern_rets.out
[ 12%] Built target syscall_intercept_base_asm
[ 13%] Building ASM object test/CMakeFiles/pattern_rets.out.dir/pattern_rets.out.s.o
[ 13%] Linking C executable hook_test
Scanning dependencies of target pattern_double_syscall.out
[ 13%] Built target check_license_executable
[ 13%] Building ASM object test/CMakeFiles/pattern_double_syscall.out.dir/pattern_double_syscall.out.s.o
Scanning dependencies of target pattern_nop_padding8.in
[ 14%] Linking ASM shared library libpattern_jmps.out.so
[ 14%] Linking ASM shared library libpattern_jmps.in.so
[ 15%] Linking ASM shared library libpattern_rets.in.so
[ 16%] Building ASM object test/CMakeFiles/pattern_nop_padding8.in.dir/pattern_nop_padding8.in.s.o
[ 17%] Linking ASM shared library libpattern_rets.out.so
[ 17%] Built target hook_test
[ 17%] Built target pattern_jmps.out
[ 18%] Linking ASM shared library libpattern_double_syscall.out.so
Scanning dependencies of target pattern_nop_padding7.in
[ 18%] Built target pattern_rets.in
[ 18%] Built target cpp_test
[ 18%] Built target pattern_jmps.in
Scanning dependencies of target pattern_nop_padding6.out
Scanning dependencies of target pattern_nop_padding9.in
Scanning dependencies of target pattern_symbol_boundary0.in
[ 19%] Building ASM object test/CMakeFiles/pattern_nop_padding7.in.dir/pattern_nop_padding7.in.s.o
Scanning dependencies of target prog_using_vfork
[ 20%] Building ASM object test/CMakeFiles/pattern_nop_padding6.out.dir/pattern_nop_padding6.out.s.o
[ 20%] Linking ASM shared library libpattern_nop_padding8.in.so
[ 20%] Building ASM object test/CMakeFiles/pattern_nop_padding9.in.dir/pattern_nop_padding9.in.s.o
[ 21%] Building C object test/CMakeFiles/prog_using_vfork.dir/prog_using_vfork.c.o
[ 22%] Building ASM object test/CMakeFiles/pattern_symbol_boundary0.in.dir/pattern_symbol_boundary0.in.s.o
[ 22%] Built target pattern_rets.out
Scanning dependencies of target pattern_loop2.out
[ 22%] Built target pattern_double_syscall.out
[ 22%] Building ASM object test/CMakeFiles/pattern_loop2.out.dir/pattern_loop2.out.s.o
Scanning dependencies of target pattern_loop2.in
[ 23%] Linking ASM shared library libpattern_nop_padding7.in.so
[ 24%] Building ASM object test/CMakeFiles/pattern_loop2.in.dir/pattern_loop2.in.s.o
[ 24%] Built target pattern_nop_padding8.in
[ 24%] Linking ASM shared library libpattern_nop_padding6.out.so
[ 24%] Linking ASM shared library libpattern_symbol_boundary0.in.so
[ 25%] Linking ASM shared library libpattern_nop_padding9.in.so
Scanning dependencies of target pattern_loop.out
/home/szychows/repo/syscall_intercept/test/pattern_loop2.out.s:59:3: error: invalid operand for instruction
                loop    0b
                ^
make[2]: *** [test/CMakeFiles/pattern_loop2.out.dir/build.make:63: test/CMakeFiles/pattern_loop2.out.dir/pattern_loop2.out.s.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1553: test/CMakeFiles/pattern_loop2.out.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 26%] Building ASM object test/CMakeFiles/pattern_loop.out.dir/pattern_loop.out.s.o
[ 26%] Built target pattern_nop_padding7.in
/home/szychows/repo/syscall_intercept/test/pattern_loop2.in.s:55:3: error: invalid operand for instruction
                loop    0b # this one as well
                ^
[ 26%] Built target pattern_nop_padding6.out
make[2]: *** [test/CMakeFiles/pattern_loop2.in.dir/build.make:63: test/CMakeFiles/pattern_loop2.in.dir/pattern_loop2.in.s.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1590: test/CMakeFiles/pattern_loop2.in.dir/all] Error 2
[ 27%] Linking C executable prog_using_vfork
[ 27%] Built target pattern_nop_padding9.in
[ 27%] Built target pattern_symbol_boundary0.in
/home/szychows/repo/syscall_intercept/test/pattern_loop.out.s:61:3: error: invalid operand for instruction
                loop    0b
                ^
make[2]: *** [test/CMakeFiles/pattern_loop.out.dir/build.make:63: test/CMakeFiles/pattern_loop.out.dir/pattern_loop.out.s.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1627: test/CMakeFiles/pattern_loop.out.dir/all] Error 2
[ 27%] Built target prog_using_vfork
make: *** [Makefile:141: all] Error 2
```

--- a/test/pattern_loop.in.s
+++ b/test/pattern_loop.in.s
@@ -44,11 +44,11 @@
 .text
 
 text_start:
-0:		xor     rax, rax
+.L0:		xor     rax, rax
 		mov     rax, 1
 		syscall
 		cmp     rax, -1
 		mov     rax, 2
 		syscall
-		loop    0b # this instruction should not be touched
+		loop    .L0 # this instruction should not be touched
 text_end:

--- a/test/pattern_loop.out.s
+++ b/test/pattern_loop.out.s
@@ -41,7 +41,7 @@
 .text
 
 text_start:
-0:		jmp     dst0
+.L0:		jmp     dst0
 		int3
 		int3
 		int3
@@ -58,5 +58,5 @@ text_start:
 		int3
 		int3
 		int3
-		loop    0b
+		loop    .L0
 text_end:

--- a/test/pattern_loop2.in.s
+++ b/test/pattern_loop2.in.s
@@ -47,10 +47,10 @@
 
 text_start:
 		xor     rax, rax # this instruction should be left unchanged
-0:		mov     rax, 1
+.L0:		mov     rax, 1
 		syscall
 		cmp     rax, -1
 		mov     rax, 2
 		syscall
-		loop    0b # this one as well
+		loop    .L0 # this one as well
 text_end:

--- a/test/pattern_loop2.out.s
+++ b/test/pattern_loop2.out.s
@@ -42,7 +42,7 @@
 
 text_start:
 		xor     rax, rax
-0:		jmp     dst0
+.L0:		jmp     dst0
 		int3
 		int3
 		int3
@@ -56,5 +56,5 @@ text_start:
 		int3
 		int3
 		int3
-		loop    0b
+		loop    .L0
 text_end:

--- a/test/pattern_nop_padding1.out.s
+++ b/test/pattern_nop_padding1.out.s
@@ -45,14 +45,14 @@
 
 text_start:
 		xor     rax, rax
-		jmp     1f    # where a nop was originally
-0:		jmp     dst0
+		jmp     .L1    # where a nop was originally
+.L0:		jmp     dst0
 		.byte   0x00
-1:		inc     rax
+.L1:		inc     rax
 		inc     rax
 		inc     rax
 		mov     rax, 1
-		jmp     0b      # where a syscall was originally
+		jmp     .L0      # where a syscall was originally
 		cmp     rax, -1
 		mov     rax, 2
 		jmp     dst1

--- a/test/pattern_nop_padding2.out.s
+++ b/test/pattern_nop_padding2.out.s
@@ -45,16 +45,16 @@
 
 text_start:
 		xor     rax, rax
-		jmp     1f
-0:		jmp     dst0
+		jmp     .L1
+.L0:		jmp     dst0
 		.byte   0x00
 		.byte   0x00
 		.byte   0x00
-1:		inc     rax
+.L1:		inc     rax
 		inc     rax
 		inc     rax
 		mov     rax, 1
-		jmp     0b
+		jmp     .L0
 		cmp     rax, -1
 		mov     rax, 2
 		jmp     dst1

--- a/test/pattern_nop_padding3.out.s
+++ b/test/pattern_nop_padding3.out.s
@@ -45,12 +45,12 @@
 
 text_start:
 		xor     rax, rax
-		jmp     1f
-0:		jmp     dst0
+		jmp     .L1
+.L0:		jmp     dst0
 		.byte   0x00
 		.byte   0x00
 		.byte   0x00
-1:		inc     rax
+.L1:		inc     rax
 		inc     rax
 		dec     rax
 		dec     rax
@@ -88,7 +88,7 @@ text_start:
 		dec     rax
 		inc     rax
 		mov     rax, 1
-		jmp     0b
+		jmp     .L0
 		cmp     rax, -1
 		mov     rax, 2
 		jmp     dst1

